### PR TITLE
Storage: LVM test tweaks and races fixes

### PR DESCRIPTION
**Pull request by tomponline**
- Fixes some race conditions.
- Adds `make race` target to enable Go's race detector.
- Fixes LVM tests on Focal.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -147,6 +147,18 @@ endif
 	CC=$(CC) go install -a -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
 	@echo "LXD built successfully"
 
+race:
```
Comment (tomponline) [inline Makefile:150]: @stgraber the `debug` and `nocache` targets don't have the `CGO_ENABLED=0` steps for `lxd-p2c` and `lxd-agent`, shall I add them?

File: Makefile [hunk 2]
```diff
@@ -147,6 +147,18 @@ endif
...
+ifeq ($(TAG_SQLITE3),)
+	@echo "Missing custom libsqlite3, run \"make deps\" to setup."
+	exit 1
+endif
+
+	go get -t -v -d ./...
+	CC=$(CC) go install -race -v -tags "$(TAG_SQLITE3)" $(DEBUG) ./...
+	CGO_ENABLED=0 go install -v -tags netgo ./lxd-p2c
```
Comment (tomponline) [inline Makefile:158]: @stgraber race detector doesn't work without cgo, so cant enable on these two steps.

File: Makefile (hunk 1 above)
Comment (stgraber) [reply Makefile:150]: Ah yeah, we should do that

File: Makefile (hunk 2 above)
Comment (stgraber) [reply Makefile:158]: Yeah, guess we should just skip on those then

**Pull request merged**